Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,1539 @@
+    const std::string expectedInstrumentName = compInfo.name(compInfo.root());
+
+    TS_ASSERT_THROWS(NexusGeometrySave::saveInstrument(
+                         m_instrument, destinationFile, DEFAULT_ROOT_PATH),
+                     std::invalid_argument &);
+  }
+
+  void test_instrument_without_sample_throws() {
+
+    auto const &instrument =
+        ComponentCreationHelper::createInstrumentWithOptionalComponents(
+            true, false, true);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*instr.first);
+
+    ScopedFileHandle fileResource("check_no_sample_throws_test_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
+
+    TS_ASSERT(compInfo.hasDetectorInfo()); // rule out throw by no detector info
+    TS_ASSERT(compInfo.hasSource());       // rule out throw by no source
+    TS_ASSERT(!compInfo.hasSample());      // verify component has no sample
+
+    TS_ASSERT_THROWS(NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                                       DEFAULT_ROOT_PATH),
+                     std::invalid_argument &);
+  }
+
+  void test_instrument_without_source_throws() {
+
+    auto const &instrument =
+        ComponentCreationHelper::createInstrumentWithOptionalComponents(
+            false, true, true);
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+    auto &compInfo = (*instr.first);
+
+    ScopedFileHandle fileResource("check_no_source_throws_test_file.hdf5");
+    auto destinationFile = fileResource.fullPath();
+
+    TS_ASSERT(compInfo.hasDetectorInfo()); // rule out throw by no detector info
+    TS_ASSERT(compInfo.hasSample());       // rule out throw by no sample
+    TS_ASSERT(!compInfo.hasSource());      // verify component has no source
+
+    TS_ASSERT_THROWS(NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                                       DEFAULT_ROOT_PATH),
+                     std::invalid_argument &);
+  }
+
+  void test_sample_not_at_origin_throws() {
+
+    auto instrument = ComponentCreationHelper::createMinimalInstrument(
+        V3D(0, 0, -10), V3D(0, 0, 2), V3D(0, 0, 10));
+    auto instr = Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+
+    ScopedFileHandle fileResource("check_nxsource_group_test_file.hdf5");
+    std::string destinationFile = fileResource.fullPath();
+
+    TS_ASSERT_THROWS(NexusGeometrySave::saveInstrument(instr, destinationFile,
+                                                       DEFAULT_ROOT_PATH),
+                     std::invalid_argument &);
+  }
+
+  /*
+ ====================================================================
+
+ NEXUS FILE FORMAT TESTS
+
+ DESCRIPTION:
+
+ The following tests document that the file format produced by saveInstrument is
+ compliant to the present Nexus standard as of the date corresponding to the
+ latest version of this document.
+
+ LIST:
```

Comment:
I wouldn't bother listing these, the test names themselves should serve as documentation.